src/fault_injector/fault_types/track_speed_limit_fault.py

```diff
@@ -3,17 +3,53 @@
 
 from src.component import Component
 from src.fault_injector.fault_types.fault import Fault, FaultConfiguration
+from src.interlocking_component.route_controller import IInterlockingDisruptor
+from src.logger.logger import Logger
+from src.wrapper.simulation_object_updating_component import (
+    SimulationObjectUpdatingComponent,
+)
+from src.wrapper.simulation_objects import Track
 
 
 class TrackSpeedLimitFault(Fault):
     """A fault affecting the speed limit of tracks."""
 
+    configuration: "TrackSpeedLimitFaultConfiguration"
+    old_speed_limit: int
+    track: Track
+    wrapper: SimulationObjectUpdatingComponent
+    interlocking: IInterlockingDisruptor
+
+    def __init__(
+        self,
+        configuration,
+        logger: Logger,
+        wrapper: SimulationObjectUpdatingComponent,
+        interlocking: IInterlockingDisruptor,
+    ):
+        super().__init__(configuration, logger)
+        self.wrapper = wrapper
+        self.interlocking = interlocking
+
     def inject_fault(self, component: Component):
         """inject TrackSpeedLimitFault into the given component
 
         :param component: The component the fault should be injected into
         :type component: Component
         """
+        self.track: Track = [
+            track
+            for track in self.configuration.wrapper_component.tracks()
+            if track.id == self.configuration.affected_element_id
+        ][0]
+        self.old_speed_limit = self.track.max_speed
+        self.track.max_speed = self.configuration.new_speed_limit
+
+        self.configuration.interlocking_component.insert_track_speed_limit_changed(
+            self.configuration.affected_element_id
```

Comment:
I don't know about @Lietze s code, but would be passing the domain object `Track` be a more robust design?